### PR TITLE
Use Twisted for VLC communication

### PR DESCRIPTION
**Pull request by Et0h**

### CODE REVIEW

File: syncplay/players/vlc.py [hunk 1]
```diff
@@ -13,11 +11,82 @@
...
+        if self.factory.connected:
+            if not self.factory.requestedVLCVersion:
+                self.factory.requestedVLCVersion = True
+                self.sendLine("get-vlc-version")
+            # try:
+            lineToSend = line.encode('utf-8') + self.delimiter
+            self.transport.write(lineToSend)
+            if self.factory._playerController._client and self.factory._playerController._client.ui:
+                self.factory._playerController._client.ui.showDebugMessage("player >> {}".format(line))
+            # except:
```
Comment (albertosottile) [inline syncplay/players/vlc.py:40]: This try/except should probably be restored.

File: syncplay/players/vlc.py [hunk 2]
```diff
@@ -13,11 +11,82 @@
...
+        elif self.vlcHasResponded:
+            self._playerController.drop()
+        else:
+            self.vlcHasResponded = True
+            self._playerController.drop(getMessage("vlc-failed-connection").format(constants.VLC_MIN_VERSION))
+
+    def closeVLC(self):
+        self._playerController._vlcclosed.set()
+        if not self.connected and not self.timeVLCLaunched:
+            # For circumstances where Syncplay is not connected to VLC and is not reconnecting
```
Comment (albertosottile) [inline syncplay/players/vlc.py:84]: Same for this one.

File: syncplay/players/vlc.py [hunk 3]
```diff
@@ -63,21 +132,18 @@ def __init__(self, client, playerPath, filePath, args):
...
-            if not self._vlcready.wait(constants.VLC_OPEN_MAX_WAIT_TIME):
-                self._vlcready.set()
-                self._client.ui.showErrorMessage(getMessage("vlc-failed-connection"), True)
-                self.reactor.callFromThread(self._client.stop, True,)
-            self.reactor.callFromThread(self._client.initPlayer, self,)
-        except:
-            pass
+
+    def initWhenConnected(self):
+        if not self._vlcready.wait(constants.VLC_OPEN_MAX_WAIT_TIME):
```
Comment (albertosottile) [inline syncplay/players/vlc.py:142]: This was also wrapped in a try/except before...

**Review by albertosottile — COMMENTED** [review]
I remember writing this a year ago... I if I recall this correctly, preliminary tests on macOS worked fine, but this MR should be extensively tested on all the other platforms. 

Also, there is this note in one of the commit messages that probably deserves an investigation:
> Known issues: there is a noticeable lag (~ 1 second) between the start of Syncplay MainWindow and the start of VLC.

File: syncplay/players/vlc.py (hunk 3 above)
Comment (Et0h) [reply syncplay/players/vlc.py]: Done all of these.

**Comment by Et0h** [comment]
> > Known issues: there is a noticeable lag (~ 1 second) between the start of Syncplay MainWindow and the start of VLC.

It's slightly faster than that for me on Windows - there is only usually a notable delay if I'm running VLC for the first time.

**Comment by Et0h** [comment]
Thanks for your work on this Alberto.

**Comment by smiba** [comment]
This may have broken VLC support for some, my friend coudn't use VLC with Syncplay v1.6.6 anymore. Syncplay would just simply not find the running VLC instance and it behaved as nothing was connected.

Downgrading to v1.6.5 fixed it

**Comment by Et0h** [comment]
@smiba Thanks for bringing this to our attention. Please provide the following information on how to reproduce this issue:
* What version of VLC was used.
* What Operating System was used.
* By what means was Syncplay installed.
* What do you mean by: "Syncplay would just simply not find the running VLC instance and it behaved as nothing was connected."

**Comment by Et0h** [comment]
@smiba Please see #365 for a discussion of this issue.

**Comment by Et0h** [comment]
The issue raised by @smiba is hopefully resolved by the fix discussed in #365.

**Pull request merged**